src/toil/lib/aws/iam.py

```diff
@@ -49,10 +49,36 @@
                                   "ec2:StartInstances",
                                   "ec2:StopInstances",
                                   "ec2:TerminateInstances",
-                                  }
+                                  ]
 
+AllowedActionCollection = Dict[str, Dict[str, List[str]]]
 
-def policy_permissions_allow(given_permissions: Set[str] = set(), required_permissions: Set[str] = set()) -> bool:
+def init_action_collection() -> AllowedActionCollection:
+    '''
+    Initialization of an action collection, an action collection contains allowed actions and not actions
+    by resource
```

Comment:
```suggestion
    by resource. Each is a pattern with wildcards, and A NotAction allows
    all actions that *don't* match the pattern.
```

Maybe explain what this concept is?